qcodes/tests/mockers/test_simulated_ats_api.py

```diff
@@ -0,0 +1,63 @@
+import pytest
+import numpy as np
+from qcodes.instrument.mockers.simulated_ats_api import SimulatedAlazarATSAPI
+from qcodes.instrument_drivers.AlazarTech.ATS import AcquisitionInterface
+from qcodes.instrument_drivers.AlazarTech.ATS9360 import (
+    AlazarTech_ATS9360 as ATS9360)
+
+
+
+@pytest.fixture(scope='function')
+def simulated_alazar():
+    driver = ATS9360(
+        'Alazar',
+        api=SimulatedAlazarATSAPI(dll_path='simulated'))
+    with driver.syncing():
+        driver.sample_rate(1_000_000_000)
+        driver.decimation(1)
+        driver.trigger_delay(0)
+    try:
+        yield driver
+    finally:
+        if driver:
+            driver.close()
+
+
+@pytest.fixture(scope='function')
+def alazar_ctrl():
+    class TestAcquisitionController(AcquisitionInterface):
+
+        def __init__(self):
+            self.buffers = []
+
+        def handle_buffer(self, buffer, buffer_number=None):
+            self.buffers.append(np.copy(buffer))
+
+        def post_acquire(self):
+            return self.buffers
+
+    yield TestAcquisitionController()
+
+
+def test_simulated_alazar(simulated_alazar, alazar_ctrl):
+    alazar = simulated_alazar
+    alazar.aux_io_mode('AUX_IN_TRIGGER_ENABLE')
+    alazar.aux_io_param('TRIG_SLOPE_POSITIVE')
+    alazar.sync_settings_to_card()
+
+    alazar.acquire(
```

Comment:
can we add an assertion of what the call to acquire returns? this would really show that the simulator works and also would test that the default buffer generator works.

Reply:
Yes that is a good suggestion!

Reply:
awesome, good catch!